VmAgent.Core/ProcessOutputLogger.cs

```diff
@@ -0,0 +1,97 @@
+{
+    using System;
+    using System.Diagnostics;
+    using global::VmAgent.Core.Dependencies.Interfaces.Exceptions;
+    using Microsoft.Azure.Gaming.VmAgent.Core.Dependencies;
+    using Microsoft.Azure.Gaming.VmAgent.Core.Dependencies.Interfaces;
+
+    public class ProcessOutputLogger: IProcessOutputLogger
+    {
+        private MultiLogger _logger;
+        private IFileWriteWrapper _fileWriteWrapper;
+
+        public ProcessOutputLogger(string logFilePath, MultiLogger logger) 
+            :this (logFilePath, logger, new FileWriteWrapper()) 
+        { 
+        }
+
+        public ProcessOutputLogger(string logFilePath, MultiLogger logger, IFileWriteWrapper fileWriteWrapper)
+        {
+            _logger = logger;
+            _fileWriteWrapper = fileWriteWrapper;
+
+            if (String.IsNullOrWhiteSpace(logFilePath))
+            {
+                throw new ProcessOuputLoggerCreationFailedException($"LogFilePath cannot be null or empty.");
+            }
+
+            try
+            {
+                _fileWriteWrapper.CreateFile(logFilePath);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogException($"Exception was thrown while creating a file. Closing a file {logFilePath}.", ex);
+                Close();
+            }
+        }
+        public void Log(string message, string streamType = "stdout")
+        {
+            try
+            {
+                _fileWriteWrapper.Write(message, streamType);
+            }
+            catch(Exception ex)
+            {
+                _logger.LogException($"Exception was thrown while writing a log. Closing a file {GetProcessLogFilePath()}.", ex);
+                Close();
+            }
+        }
+
+        public string GetProcessLogFilePath()
+        {
+            return _fileWriteWrapper.GetProcessLogFilePath();
+        }
+
+        public void Close()
+        {
+            _logger.LogInformation($"Closing a Process Log file... Target File Path : {GetProcessLogFilePath()}. ");
+
+            try
+            {
+                if (_fileWriteWrapper.Close())
+                {
+                    _logger.LogInformation($"File is successfully closed.");
+                }
+                else
+                {
+                    _logger.LogInformation($"File is already closed.");
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogException($"Exception was thrown while closing a file..", ex);
+            }
+        }
+
+        public void ErrorOutputHandler(object sendingProcess, DataReceivedEventArgs outLine)
+        {
+            Log(outLine.Data, "stderr");
+        }
+
+        public void StdOutputHandler(object sendingProcess, DataReceivedEventArgs outLine)
+        {
+            Log(outLine.Data);
+        }
+
+        public void ProcessExitedHanlder(object sendingProcess, EventArgs outLine)
+        {
+            Log("Process is terminated.");
```

Comment:
I'm not sure this log is necessary. I guess it can be used to show that the process exited cleanly? The wording here can also lead to a bit of confusion--is the process terminating on its own or has it been terminated by an external force?